app/assets/src/components/views/samples/ProjectSettingsModal/UserManagementForm.jsx

```diff
@@ -0,0 +1,130 @@
+import React from "react";
+import PropTypes from "prop-types";
+import PrimaryButton from "~ui/controls/buttons/PrimaryButton";
+import { Input } from "~ui/controls";
+import StringHelper from "~/helpers/StringHelper";
+import cs from "./user_management_form.scss";
+import cx from "classnames";
+import axios from "axios";
+import UserIcon from "../../../ui/icons/UserIcon";
+
+class UserManagementForm extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      username: "",
+      email: "",
+      statusMessage: "",
+      statusClass: null
+    };
+  }
+
+  handleAddUser = () => {
+    const { csrf, onUserAdded, project } = this.props;
+    const { username, email } = this.state;
+
+    if (StringHelper.validateEmail(email) && StringHelper.validateName(name)) {
+      this.setState({
+        statusClass: cs.infoMessage,
+        statusMessage: "Sending invitation"
+      });
+      axios
+        .put(`/projects/${project.id}}/add_user`, {
+          user_name_to_add: username,
+          user_email_to_add: email,
+          authenticity_token: csrf
+        })
+        .then(() => {
+          onUserAdded({ username, email });
+          this.setState({
+            inviteStatus: "Invitation sent! User added."
+          });
+        });
+    } else {
+      this.setState({
+        statusClass: cs.errorMessage,
+        statusMessage: "Invalid name or email address"
+      });
+    }
+  };
+
+  handleChangeUsername = username => {
+    this.setState({
+      username,
+      statusClass: cs.infoMessage,
+      statusMessage: ""
+    });
+  };
+
+  handleChangeEmail = email => {
+    this.setState({
+      email,
+      statusClass: cs.infoMessage,
+      statusMessage: ""
+    });
+  };
+
+  render() {
+    const { users } = this.props;
+
+    return (
+      <div className={cs.userManagementForm}>
+        <div className={cs.title}>
+          <UserIcon className={cs.icon} />
+          Project Members
+        </div>
+        <div className={cs.userList}>
+          {users.length > 0 ? (
+            users.map(user => {
+              return (
+                <div className={cs.userEntry} key={user.email}>
+                  {user.name} ({user.email})
+                </div>
+              );
+            })
+          ) : (
+            <div className={cs.userEntry} key="None">
+              None
+            </div>
+          )}
+        </div>
+        <div className={cs.fillIn} />
+        <div className={cx(cs.statusMessage, this.state.statusClass)}>
+          {this.state.statusMessage}
+        </div>
+        <div className={cs.addMemberForm}>
+          <div className={cs.addMemberFormField}>
```

Comment:
I think this UI needs another look.

At the moment the Inputs are unlabeled, so it's not clear what to put in them. Can add a label or a placeholder.

Also not clear why the user needs to enter both a name AND an email to add a user. When I looked at the back-end, I realized that the back-end adds the existing user if the e-mail already exists, and creates a new user if not.

We might consider splitting up the UI to handle "Add existing user" and "Add new user" separately? Maybe talk to Jenn

Reply:
Your concern on existing vs. new users is valid but is out of scope of this change. Existing user for instance should have a search field or something where you pick the user.

This PR does not change the current functionality on that. Eventually, we can add that, but not the focus of this change. Created JIRA ticket for that: https://jira.czi.team/browse/IDSEQ-834.